spring-kafka/src/main/java/org/springframework/kafka/support/TopicPartitionInitialOffset.java

```diff
@@ -80,9 +98,28 @@ public TopicPartitionInitialOffset(String topic, int partition, Long initialOffs
 	 * @since 1.1
 	 */
 	public TopicPartitionInitialOffset(String topic, int partition, Long initialOffset, boolean relativeToCurrent) {
+		this(topic, partition, initialOffset, relativeToCurrent, null);
+	}
+
+	/**
+	 * Construct an instance with the provided initial offset.
+	 * @param topic the topic.
+	 * @param partition the partition.
+	 * @param initialOffset the initial offset.
+	 * @param relativeToCurrent true for the initial offset to be relative to
+	 * the current consumer position, false for a positive initial offset to
+	 * be absolute and a negative offset relative to the current end of the
+	 * partition.
+	 * @param position {@link SeekPosition#BEGINNING} or {@link SeekPosition#END}
+	 * overrides initialOffset and relativeToCurrent.
```

Comment:
If that is a fact, why just don't encourage end-user to type those extra options?
In other words if they are so mutually exclusive, let's make a ctor without them!
```
public TopicPartitionInitialOffset(String topic, int partition, Long initialOffset, boolean relativeToCurrent)
...
public TopicPartitionInitialOffset(String topic, int partition, SeekPosition position) 
```